internal/service/dms/replication_task.go

```diff
@@ -373,9 +382,14 @@ func startReplicationTask(id string, conn *dms.DatabaseMigrationService) error {
 		return fmt.Errorf("error reading DMS Replication Task (%s): empty output", id)
 	}
 
+	startReplicationTaskType := dms.StartReplicationTaskTypeValueStartReplication
+	if fromStatus != replicationTaskStatusReady {
+		startReplicationTaskType = dms.StartReplicationTaskTypeValueResumeProcessing
+	}
+
 	_, err = conn.StartReplicationTask(&dms.StartReplicationTaskInput{
 		ReplicationTaskArn:       task.ReplicationTaskArn,
-		StartReplicationTaskType: aws.String(dms.StartReplicationTaskTypeValueStartReplication),
+		StartReplicationTaskType: aws.String(startReplicationTaskType),
```

Comment:
Author's note: This change is necessary because for tasks of type `full-load` and `full-load-and-cdc` [you can only use `start-replication` when it's the first time the task is run](https://docs.aws.amazon.com/dms/latest/APIReference/API_StartReplicationTask.html#DMS-StartReplicationTask-request-StartReplicationTaskType). You _have_ to use `resume-processing`, otherwise it will fail.

Proof of it, is the following error while running the acceptance tests without this change:
```
replication_task_test.go:100: Step 4/4 error: Error running apply: exit status 1

Error: error starting DMS Replication Task (tf-acc-test-748333755630122031): InvalidParameterCombinationException: Start Type : START_REPLICATION, valid only for tasks running for the first time status code: 400, request id: af7935aa-89d8-49de-b835-8d6a87addc65

with aws_dms_replication_task.test, on terraform_plugin_test.tf line 176, in resource "aws_dms_replication_task" "test":
    176: resource "aws_dms_replication_task" "test" {

--- FAIL: TestAccDMSReplicationTask_startReplicationTask (1373.44s)
```.

This way, we _only_ use `start-replication` after the task is on status `ready`, which [only happens right after the task has been created](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Monitoring.html#CHAP_Tasks.Status).